homeassistant/components/device_tracker/asuswrt.py

```diff
@@ -112,12 +115,17 @@ def __init__(self, config):
         self.ssh_key = config.get('ssh_key', config.get('pub_key', ''))
         self.protocol = config[CONF_PROTOCOL]
         self.mode = config[CONF_MODE]
+        self.port = config.get(CONF_PORT, 22)
+        self.ssh_args = {}
 
         if self.protocol == 'ssh':
+
+            if self.port != DEFAULT_SSH_PORT:
+                self.ssh_args['port'] = self.port
```

Comment:
I agree. I think when I made this change a couple months ago, I did so because the asus_wrt tests were really picky about the port, so I was trying to keep the port 22 when nothing was specified. I was trying (unsuccessfully) to get the tests to pass. I have since figured out the tests, so I'll make this change.